### PR TITLE
Test scoped logs in wait functions

### PR DESCRIPTION
**Pull request by pierDipi**
Test logged using `log.P*` are not straightforward to find for debugging issues
so, I'm passing `feature.T` to `Wait*` functions.

# Changes

- Test scoped logs in wait functions

/kind bug

Fixes #

**Release Note**

```release-note
Add feature.T parameter to Wait* functions for test-scoped logging.
```

**Docs**

```docs

```

### CODE REVIEW

**Comment by pierDipi** [comment]
Abusing downstream tests to see which repo breaks.

**Comment by pierDipi** [comment]
I'll fix downstream repositories once this PR is in.

**Comment by pierDipi** [comment]
Ping @devguyio @lionelvillard @evankanderson 
This is preventing eventing from a proper debug story for rekt tests.

File: pkg/k8s/wait.go [hunk 1]
```diff
@@ -169,9 +169,9 @@ func isReady(name string) ConditionFunc {
 		allReady := true
 		for _, c := range obj.Status.Conditions {
 			if !c.IsTrue() {
-				msg := fmt.Sprintf("%s is not %s, %s: %s", name, c.Type, c.Reason, c.Message)
+				msg := fmt.Sprintf("%s is not %s\n\nResource: %s\n", name, ready.Type, resource(obj))
```
Comment (devguyio) [inline pkg/k8s/wait.go:172]: Is there a reason you're no longer logging the individual condition that's not true? you changed `c.Type` to `ready.Type` maybe that's just cope pasted from the previous logging?

**Review by devguyio — COMMENTED** [review]
Thanks for doing this, left one comment.

File: pkg/k8s/wait.go (hunk 1 above)
Comment (pierDipi) [reply pkg/k8s/wait.go:172]: Yeah, no good reason just copy-paste.

```suggestion
				msg := fmt.Sprintf("%s is not %s, %s: %s\n\nResource: %s\n", name, c.Type, c.Reason, c.Message, resource(obj))
```

Comment (pierDipi) [reply pkg/k8s/wait.go]: Done

**Review by devguyio — APPROVED** [review]
/lgtm
/approve

**Pull request merged**